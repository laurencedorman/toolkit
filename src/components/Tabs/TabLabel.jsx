/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

/* eslint-disable */
// TODO in https://manomano.atlassian.net/browse/REACT-243

import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import styles from './Tabs.module.scss';

const TabLabel = ({ activeTab, className, label, onClick }) => {
  const handleClick = () => {
    onClick(label);
  };

  const classNames = cn('toolkit', className, {
    [styles.active]: activeTab === label,
  });

  return (
    <li
      className={styles.tabLabel}
      aria-selected={activeTab === label}
      role="tab"
    >
      <span className={classNames} onClick={handleClick}>
        {label}
      </span>
    </li>
  );
};

TabLabel.displayName = 'TabLabel';

TabLabel.propTypes = {
  label: PropTypes.node.isRequired,
  activeTab: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
};

TabLabel.defaultProps = {
  className: '',
  activeTab: '',
};

export default TabLabel;
