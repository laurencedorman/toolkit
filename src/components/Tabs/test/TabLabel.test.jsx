/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import TabLabel from '../TabLabel';

describe('TabLabel', () => {
  const onClickMock = jest.fn();
  const mockLabel = 'test';

  const wrapper = shallow(
    <TabLabel onClick={onClickMock} label={mockLabel} activeTab="test" />
  );

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('span click invoke onClick props', () => {
    wrapper.find('span').simulate('click');
    expect(onClickMock).toHaveBeenCalledWith(mockLabel);
  });
});
