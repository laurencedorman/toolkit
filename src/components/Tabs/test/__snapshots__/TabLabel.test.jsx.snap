// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabLabel should match snapshot 1`] = `
<li
  aria-selected={true}
  className="tabLabel"
  role="tab"
>
  <span
    className="toolkit active"
    onClick={[Function]}
  >
    test
  </span>
</li>
`;
